lighthouse-core/scripts/update-report-fixtures.js

```diff
@@ -43,15 +47,28 @@ async function update() {
     res(address.port);
   }));
 
+  const oldArtifacts = await assetSaver.loadArtifacts(artifactPath);
+
   const url = `http://localhost:${port}/dobetterweb/dbw_tester.html`;
   const rawFlags = [
-    '--gather-mode=lighthouse-core/test/results/artifacts',
+    `--gather-mode=${artifactPath}`,
     '--throttling-method=devtools',
     url,
   ].join(' ');
   const flags = cliFlags.getFlags(rawFlags);
   await cli.runLighthouse(url, flags, budgetedConfig);
   await new Promise(res => server.close(res));
+
+  if (artifactName) {
+    // Revert everything except the one artifact
+    const newArtifacts = await assetSaver.loadArtifacts(artifactPath);
+    if (!(artifactName in newArtifacts) && !(artifactName in oldArtifacts)) {
+      throw Error('Unknown artifact name: ' + artifactName);
```

Comment:
```suggestion
      console.warn(`❌ Unknown artifact name: '${artifactName}'. Reverting artifacts...`); // eslint-disable-line no-console
```
throwing ends up as an unhandled promise rejection (since the function is async) and leaves the artifacts as the newly generated ones. With this it logs and continues. Since the `artifactName` is unknown, it's just setting the property to `undefined` which doesn't change the JSON output and so the original artifacts are restored.